backend/tests/routes.test.js

```diff
@@ -0,0 +1,54 @@
+const request = require("supertest");
+const app = require("../index");
+const group_id = "84582b22-2463-4415-b992-bcccc5354301";
+
+describe("Post Group", () => {
+  it("should create a new group", async () => {
+    const res = await request(app)
+      .post("/api/groups")
+      .send({
+        name: "group1",
+        creationDate: "2020-06-30 15:39:52.270000",
+        createdBy: "dbbd63b8-b481-466b-bb1a-d0e2b43d8afd",
+        claims: ["claim1", "claim2"],
+        advancedSettings: { key: "key", value: "value" },
+      });
+    expect(res.statusCode).toEqual(200);
+  });
+});
+
+describe("Get Group by uuid", () => {
+  it("should get a group by uuid", async () => {
+    const res = await request(app).get(`/api/groups/${group_id}`);
+    expect(res.statusCode).toEqual(200);
+  });
+});
+
+describe("Get all Groups", () => {
+  it("should get all groups", async () => {
+    const res = await request(app).get(`/api/groups`);
+    expect(res.statusCode).toEqual(200);
+  });
+});
+
+describe("Update Group", () => {
+  it("should update a group", async () => {
+    const res = await request(app)
+      .patch(`/api/groups/update/${group_id}`)
+      .send({
+        name: "group1",
+        creationDate: "2020-06-30 15:39:52.270000",
+        createdBy: "dbbd63b8-b481-466b-bb1a-d0e2b43d8afd",
+        claims: ["claim1", "claim2"],
+        advancedSettings: { key: "key", value: "value" },
+      });
+    expect(res.statusCode).toEqual(200);
+  });
+});
+
+describe("Delete Group", () => {
+  it("should delete a group", async () => {
+    const res = await request(app).delete(`/api/groups/delete/${group_id}`);
+    expect(res.statusCode).toEqual(200);
+  });
+});
```

Comment:
You should add negative cases also. For instance:

-  create a group without giving the name. What is happening in this case? 

- get a group by invalid groupId, or by a groupId which is not an uuid

- update a group with an invalid value

- delete a group which does not exists